backend/remote-state/s3/backend.go

```diff
@@ -155,20 +155,23 @@ func New() backend.Backend {
 				Optional:    true,
 				Description: "Skip getting the supported EC2 platforms.",
 				Default:     false,
+				Deprecated:  "This attribute is no longer used.",
```

Comment:
Is there any more context to this? e.g. why are we deprecating this and what do we expect users which used this in the past to do? As far as I remember this also had some performance background.

Regardless: Do we need to introduce deprecations as part of this PR which is swapping a library?

Reply:
Ah yes. 😄 The reason for the deprecation is that for the S3 Backend this EC2 service functionality is not needed since we are only interfacing with the S3 service and potentially the DynamoDB service. When I split off the new shared library, I specifically omitted it from the library since only the Terraform AWS Provider actually needs it. The Terraform AWS Provider will continue to utilize the functionality and provide that configuration option.

Reply:
👌Removing `skip_get_ec2_platforms` makes sense. I'd just add the explanation there right to the deprecation message, so it's clear.

I'm not sure about the other two though:
 - `skip_region_validation` - this provider still uses region which may get validated?
 - `skip_requesting_account_id` - there's a pending PR already which is basically making this field relevant https://github.com/hashicorp/terraform/pull/20278

Reply:
> `skip_region_validation` - this provider still uses region which may get validated?

The utility of the region validation check is questionable, in my opinion. There are valid cases where the validation gets in the way:

* AWS turns on new region in Standard/GovCloud/China partition: Operators have to wait for a Terraform release with updated AWS Go SDK that captures the updated region and know to implement the workaround in the meantime. This happened recently with `eu-north-1`.
* AWS has valid regions outside Standard/GovCloud/China partitions (e.g. AWS Secret and AWS Top Secret partitions) which are not captured by the default region validation.

I'm not heavily opposed to adding the check into the new library and still validating it with the S3 Backend though.

> * `skip_requesting_account_id` - there's a pending PR already which is basically making this field relevant #20278

Previously requesting account ID did nothing useful for the S3 Backend beyond additional IAM/STS API calls which could fail, so the removal of the account ID logic from the S3 Backend provides adds stability.

In the context of the feature request mentioned and this argument, enabling the `skip_requesting_account_id` argument and either of the new `allowed_account_ids`/`forbidden_account_ids` arguments seems like they would represent a conflicting configuration that should error. Rather than potentially having conflicting arguments, it seems like we can tune that pull request functionality to only do the account ID lookup when the new arguments are defined.

I think this represents a useful split in functionality from the Terraform AWS Provider, where the Terraform AWS Provider actually requires the account ID lookup for various manual ARN construction, while the S3 Backend would only need it in the context of optionally validating the account ID itself.

Reply:
Re `skip_region_validation`

Thanks a lot for describing the background here. It sounds like keeping this option would allow folks to use (and keep using) new and secret regions, which I think is desirable? In other words it is desirable that we keep the option for these people?

I don't have too strong opinion about this one though and block the PR on it, but I'd be more inclined to keep this one.

> Rather than potentially having conflicting arguments, it seems like we can tune that pull request functionality to only do the account ID lookup when the new arguments are defined.

That's a great idea 👍 Feel free to deprecate it then.